service/src/main/java/com/derpgroup/quip/manager/QuipManager.java

```diff
@@ -101,47 +119,56 @@ protected void doGoodbyeRequest(VoiceInput voiceInput,
   protected void doConversationRequest(VoiceInput voiceInput,
       SsmlDocumentBuilder builder) {
 
+    Map<String,String> messageMap = voiceInput.getMessageAsMap();
     QuipMetadata metadata = (QuipMetadata) voiceInput.getMetadata();
-
     String messageSubject = voiceInput.getMessageSubject();
 
+    switchOnSubject(messageSubject, messageMap, builder, metadata);
+  }
+
+  public void switchOnSubject(String messageSubject, Map<String,String> messageMap, SsmlDocumentBuilder builder, QuipMetadata metadata) {
     switch (messageSubject) {
     case "COMPLIMENT":
-      doComplimentRequest(voiceInput, builder, metadata);
+      doComplimentRequest(messageMap, builder, metadata);
       break;
     case "INSULT":
-      doInsultRequest(voiceInput, builder, metadata);
+      doInsultRequest(messageMap, builder, metadata);
       break;
     case "BACKHANDED_COMPLIMENT":
-      doBackhandedComplimentRequest(voiceInput, builder, metadata);
+      doBackhandedComplimentRequest(messageMap, builder, metadata);
       break;
     case "WINSULT":
-      doWinsultRequest(voiceInput, builder, metadata);
-      break;
-    case "HELP":
```

Comment:
We need HELP back in here as a switchable subject, else the "another" request can't trigger it to replay. And it blows up as:
"Unknown request type amazon.helpintent"